File: Ejes Codeforces/Resueltos/A. Double Cola.py
from sys import stdin

def person(n) -> int:
    
    

def main():
    line = stdin.readline().strip()
    while len(line) != 0:
        n = person(int(line))
        if n == 0:
            print("Sheldon")
        elif n == 1:
            print("Leonard")
        elif n == 2:
            print("Penny")
        elif n == 3:
            print("Rajesh")
        elif n == 4:
            print("Howard")
        line = stdin.readline().strip()

main()

File: codeforces/Resueltos/A. Double Cola.cpp
#include <cstdio>
#include <cstring>
#include <cmath>

using namespace std;

int calc(int x) {
    if (1 <= x && x <= 5) return x;

    int c = ceil( log((double)x/5 +1) / log((double)2) ) -1;

    x -= 5*((1 << c) -1);

    return ceil( (double)x / (double)(1 << c) );
}

int main() {
  int n;
  scanf("%d", &n);

  char s[10];
  
  switch(calc(n)) {
  case 1:
    strcpy(s, "Sheldon");
    break;
  case 2:
    strcpy(s, "Leonard");
    break;
  case 3:
    strcpy(s, "Penny");
    break;
  case 4:
    strcpy(s, "Rajesh");
    break;
  case 5:
    strcpy(s, "Howard");
    break;
  }

  printf("%s\n", s);

  return 0;
}
